src/meta/test/BalancerTest.cpp

```diff
@@ -912,21 +887,18 @@ TEST(BalanceTest, StopPlanTest) {
           Return(ByMove(folly::makeFuture<Status>(Status::OK()).delayed(std::chrono::seconds(3)))))
       .WillOnce(
           Return(ByMove(folly::makeFuture<Status>(Status::OK()).delayed(std::chrono::seconds(3)))));
+
   FLAGS_task_concurrency = 8;
   JobDescription jd = makeJobDescription(space, kv, cpp2::JobType::DATA_BALANCE);
   ZoneBalanceJobExecutor balancer(jd, kv, &delayClient, {});
   balancer.spaceInfo_.loadInfo(space, kv);
-  balancer.lostZones_ = {"4", "5"};
-  folly::Baton<true, std::atomic> baton;
-  balancer.setFinishCallBack([&](meta::cpp2::JobStatus) {
-    baton.post();
-    return nebula::cpp2::ErrorCode::SUCCEEDED;
-  });
-  auto ret = balancer.executeInternal();
-  EXPECT_EQ(nebula::cpp2::ErrorCode::SUCCEEDED, ret.value());
+  balancer.lostZones_ = {"5"};
+
+  auto ret = balancer.executeInternal().get();
```

Comment:
Sorry for late response, would you try to don't call in sync here. We could call `ret = balancer.executeInternal();` and `balancer.stop()`. Then call `ret.wait()`.

This test is supposed to test stop the running tasks, if modify in this way, it is meaningless actually.

Reply:
Has fixed.